pydantic_ai_slim/pydantic_ai/models/bedrock.py

```diff
@@ -414,22 +417,38 @@ async def _map_messages(
                                     ],
                                 }
                             )
-            elif isinstance(m, ModelResponse):
+            elif isinstance(message, ModelResponse):
                 content: list[ContentBlockOutputTypeDef] = []
-                for item in m.parts:
+                for item in message.parts:
                     if isinstance(item, TextPart):
                         content.append({'text': item.content})
                     else:
                         assert isinstance(item, ToolCallPart)
                         content.append(self._map_tool_call(item))
                 bedrock_messages.append({'role': 'assistant', 'content': content})
             else:
-                assert_never(m)
+                assert_never(message)
+
+        # Merge together sequential user messages.
+        processed_messages: list[MessageUnionTypeDef] = []
+        last_message: dict[str, Any] | None = None
```

Comment:
Can we type this as `MessageUnionTypeDef` instead of an arbitrary dict?

Reply:
Not easily. When I make that change I get the error below. Attempting to fix that error leads to other errors and in the end I don't know how deep the rabbit hole will be nor how complicated the code will get. This solution keeps the code compact and reasonably easy to follow.

```
Could not assign item in TypedDict
  "list[ContentBlockUnionTypeDef]" is not assignable to "list[ContentBlockOutputTypeDef]"
    Type parameter "_T@list" is invariant, but "ContentBlockUnionTypeDef" is not the same as "ContentBlockOutputTypeDef"
    Consider switching from "list" to "Sequence" which is covariant
```